syncer/db.go

```diff
@@ -69,45 +69,92 @@ type binlogSize struct {
 	size int64
 }
 
-// Conn represents a live DB connection
-type Conn struct {
-	cfg      *config.SubTaskConfig
-	baseConn *baseconn.BaseConn
+func createBaseDB(dbCfg config.DBConfig) (*conn.BaseDB, error) {
+	db, err := conn.DefaultDBProvider.Apply(dbCfg)
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
 }
 
-// ResetConn reset baseConn.*DB's connection pool
-func (conn *Conn) ResetConn(tctx *tcontext.Context) error {
-	if conn.baseConn == nil {
-		return terror.ErrDBUnExpect.Generate("database base connection not valid")
+// close baseDB to release all connection generated by this baseDB and this baseDB
+func closeBaseDB(tctx *tcontext.Context, baseDB *conn.BaseDB) {
+	if baseDB != nil {
+		err := baseDB.Close()
+		if err != nil {
+			tctx.L().Error("fail to close baseDB", log.ShortError(err))
+		}
 	}
-	return conn.baseConn.ResetConn(tctx)
 }
 
-func (conn *Conn) getMasterStatus(flavor string) (mysql.Position, gtid.Set, error) {
-	return utils.GetMasterStatus(conn.baseConn.DB, flavor)
+// UpStreamConn connect to upstream DB, use *sql.DB instead of *sql.Conn
```

Comment:
you can leave the reason here.

Reply:
ok, add [reason](https://github.com/pingcap/dm/pull/266/files#diff-c3cc6fccd12f8e1679a575d1aaf8fd14R91)